File: DSA/NeeravKumarSharma_590018263/Day_10/Q2.java
import java.util.HashSet;

class Q2 {
    public void checkDistinctCharacters(String s) {
        HashSet<Character> uniqueChars = new HashSet<>();
        
        for (char c : s.toCharArray()) {
            uniqueChars.add(c);
        }
        
        if (uniqueChars.size() % 2 == 0) {
            System.out.println("CHAT WITH HER!");
        } else {
            System.out.println("IGNORE HIM!");
        }
    }

    public static void main(String[] args) {
        Q2 obj = new Q2();
        
        System.out.print("Input: abc. Output: ");
        obj.checkDistinctCharacters("abc"); 
        
        System.out.print("Input: abcd. Output: ");
        obj.checkDistinctCharacters("abcd"); 
    }
}

/*class Solution {
    public void checkDistinctCharacters(String s) {
        boolean[] seen = new boolean[256];
        int distinctCount = 0;
        
        for (char c : s.toCharArray()) {
            if (!seen[c]) {
                seen[c] = true;
                distinctCount++;
            }
        }
        
        if (distinctCount % 2 == 0) {
            System.out.println("CHAT WITH HER!");
        } else {
            System.out.println("IGNORE HIM!");
        }
    }

    public static void main(String[] args) {
        Solution obj = new Solution();
        
        System.out.print("Input: abc. Output: ");
        obj.checkDistinctCharacters("abc"); 
        
        System.out.print("Input: abcd. Output: ");
        obj.checkDistinctCharacters("abcd"); 
    }
}*/

File: DSA/Aarush_590017879/Day_10/Q2.c
#include <stdio.h>
#include <string.h>

int main() {
    char str[101];
    scanf("%s", str);

    int freq[26] = {0};
    int distinct = 0;

    for (int i = 0; str[i] != '\0'; i++) {
        if (freq[str[i] - 'a'] == 0) {
            freq[str[i] - 'a'] = 1;
            distinct++;
        }
    }

    if (distinct % 2 == 0)
        printf("CHAT WITH HER!");
    else
        printf("IGNORE HIM!");

    return 0;
}